src/sage/matroids/circuits_matroid.pyx

```diff
@@ -217,19 +218,52 @@ cdef class CircuitsMatroid(Matroid):
             ...
             ValueError: no circuit in independent set
         """
-        cdef set I = set(F)
-        for C in self._C:
-            if C <= I:
-                return C
+        cdef set XX = set(X)
+        cdef int i, l = len(XX)
+        for i in sorted(self._k_C):
```

Comment:
It might be good to store (perhaps lazily?) the `sorted(self._k_C)`?

Reply:
Yes, at initialization.